packages/calling-stateful-client/src/StreamUtils.ts

```diff
@@ -30,210 +31,214 @@ export type CreateViewResult = {
   view: VideoStreamRendererView;
 };
 
-async function createViewRemoteVideo(
+async function createViewVideo(
   context: CallContext,
   internalContext: InternalCallContext,
   callId: string,
-  participantId: CommunicationIdentifierKind | string,
-  stream: RemoteVideoStreamState,
+  stream?: RemoteVideoStreamState | LocalVideoStreamState,
+  participantId?: CommunicationIdentifierKind | string,
   options?: CreateViewOptions
 ): Promise<CreateViewResult | undefined> {
-  // Render RemoteVideoStream that is part of a Call
-  const streamId = stream.id;
-  const streamType = stream.mediaStreamType;
-  let participantKey;
-  if (typeof participantId === 'string') {
-    participantKey = participantId;
-  } else {
-    participantKey = toFlatCommunicationIdentifier(participantId);
-  }
-  const streamLogInfo = { callId, participantKey, streamId, streamType };
-
-  _logCreateRemoteStreamEvent(EventNames.CREATING_REMOTE_VIEW, streamLogInfo);
-  const renderInfo = internalContext.getRemoteRenderInfoForParticipant(callId, participantKey, streamId);
-
-  if (!renderInfo) {
-    _logCreateRemoteStreamEvent(EventNames.REMOTE_STREAM_NOT_FOUND, streamLogInfo);
-    console.warn('RemoteVideoStream not found in state');
-    return;
-  }
-
-  if (renderInfo.status === 'Rendered') {
-    _logCreateRemoteStreamEvent(EventNames.REMOTE_STREAM_ALREADY_RENDERED, streamLogInfo);
-    console.warn('RemoteVideoStream is already rendered');
-    return;
-  }
-
-  if (renderInfo.status === 'Rendering') {
-    // Do not log to console here as this is a very common situation due to UI rerenders while
-    // the video rendering is in progress.
-    _logCreateRemoteStreamEvent(EventNames.REMOTE_STREAM_RENDERING, streamLogInfo);
-    return;
-  }
+  // we can only have 3 types of createView
+  let createViewType: 'local' | 'remote' | 'unparented';
```

Comment:
this is only set once, consider making constant and set the value at the time of declaration